### DISCUSSION TITLE
Flaky builds in CI with kroki.io

### DISCUSSION THREAD

**Discussion by l-zeuch** [thuliteio/doks#1261] [Q&A] [answered]
Hello!

We're seeing some rather temperamental builds when trying to build our site in a CI environment, specifically the following error:

```
ERROR The "kroki" code block render hook was unable to get the remote diagram. See "/home/runner/work/yagpdb-docs-v2/yagpdb-docs-v2/content/docs/reference/custom-interactions.md:38:1". failed to fetch remote resource: Bad Request
ERROR The "kroki" code block render hook was unable to get the remote diagram. See "/home/runner/work/yagpdb-docs-v2/yagpdb-docs-v2/content/docs/reference/custom-interactions.md:424:1". failed to fetch remote resource: Bad Request
```
See https://github.com/botlabs-gg/yagpdb-docs-v2/actions/runs/10122112912/job/27993831309.

The mentioned file / lines listed below, should they have syntax errors:
- [custom-interactions.md:38](https://github.com/botlabs-gg/yagpdb-docs-v2/blob/master/content/docs/reference/custom-interactions.md?plain=1#L38-L49)
- [custom-interactions.md:424](https://github.com/botlabs-gg/yagpdb-docs-v2/blob/master/content/docs/reference/custom-interactions.md?plain=1#L424-L449)

As you can see from the linked actions log, we've just introduced a retry action to make an attempt at working around the problem, to no avail.
This error does not appear when building locally, or we haven't encountered it thus far. We're also considering spinning up our own kroki docker container, though we'd prefer not to delve further than necessary into Doks' internals (we're already overriding a fair bit).

If you need more information / logs, do let me know!

**Comment by l-zeuch** [comment, marked as answer]
Follow-Up:
We managed to (finally) reproduce this in a local environment, thereby getting a usable error out of Kroki, which lead us to [kroki returns 400 on Mermaid diagrams](https://kroki.zulipchat.com/#narrow/stream/232085-users/topic/kroki.2Eio.20returns.20400.20on.20Mermaid.20diagrams) on Kroki's support channel.

We now checked the generated SVGs into our worktree, as [recommended by your documentation](https://getdoks.org/docs/built-ins/diagrams/#performance). Somehow that section elided our eyes when setting up the page.
With that change, we now have reliable builds.

Thanks for your work on this theme, our userbase really likes how polished it all looks!

**Reply by h-enk** [reply]
Thanks! I'm glad you got it working in the end
